src/test/scala/strawman/collection/test/Test.scala

```diff
@@ -288,6 +288,15 @@ class StrawmanTest {
     println(xs17.to(List))
   }
 
+  def mapOps(xs: strawman.collection.Map[Int, String]): Unit = {
+    // val xs1 = xs.map { case (k, v) => (v, k) }
+    // val xs2: strawman.collection.Map[String, Int] = xs1
+    val xs1 = xs.map(kv => (kv._2, kv._1))
+    val xs2: strawman.collection.Iterable[(String, Int)] = xs1
```

Comment:
Oh, I just realized that I defined the `map` overload that returns a `Map` with the following signature:

~~~ scala
def map[K2, V2](f: (K, V) => (K2, V2)): Map[K2, V2]
~~~

So, it takes as parameter a function with two parameters, not a function with one (tuple) parameter. But at call site I wrote code that was supplying a function taking one tuple parameter. That’s why the type inference didn’t result to what I expected.

I changed the call site and now the overloading trick works.